pkg/daemon/daemon.go

```diff
@@ -577,17 +581,14 @@ func (dn *Daemon) nodeHasAnnotation(annoKey string, value string) bool {
 	return false
 }
 
+// isNodeDraining: check if the node is draining
+// both Draining and MCP paused labels will return true
 func (dn *Daemon) isNodeDraining() bool {
 	anno, ok := dn.node.Annotations[annoKey]
 	if !ok {
 		return false
 	}
 
-	if dn.openshiftContext.IsOpenshiftCluster() && !dn.openshiftContext.IsHypershift() {
-		// for openshift cluster draining should be true only if the annotation has MCP paused
-		return anno == annoMcpPaused
-	}
-
 	return anno == annoDraining || anno == annoMcpPaused
```

Comment:
The assumption is that this function is only called in openshift?

Reply:
No that is why we add the or.

The function will return true if one of the annotations exist.

We use this to continue the work after the node or the config daemon gets reset.

Because another node takes the leader election so you don't want to try and get the lock again we just continue